python/google/protobuf/internal/reflection_test.py

```diff
@@ -1505,6 +1508,24 @@ def testMergeFromBug(self):
     message2.MergeFrom(message1)
     self.assertFalse(message2.HasField('optional_nested_message'))
 
+  def testReflectionDoesntBreakMerging(self):
```

Comment:
Add the test to message_factory_test instead.

Reply:
Done.